### PR TITLE
Best rated products selector and list

### PR DESCRIPTION
**Pull request by alexWaligorski**

### CODE REVIEW

File: src/features/ShopHomePage.js [hunk 1]
```diff
@@ -11,7 +15,36 @@ export default function ShopHomePage() {
...
-  if (isLoadingCategories || isLoadingProducts) {
+  const { data: bestRatedProducts, isLoading: isLoadingBestRated } =
+    useGetAllProductsQuery(undefined, {
+      selectFromResult: ({ data, isLoading, isUninitialized }) => {
+        const filteredData = (data ?? []).filter(
+          (product) => product.rating > 4.9
+        );
+        return {
+          data: filteredData,
+          isLoading: isUninitialized || isLoading,
```
Comment (woeishi) [inline src/features/ShopHomePage.js:26]: add issue link for later, about why this isLoading hack is necessary

File: src/features/ShopHomePage.js [hunk 2]
```diff
@@ -11,7 +15,36 @@ export default function ShopHomePage() {
...
+        };
+      },
+    });
+
+  console.log(bestRatedProducts);
```
Comment (woeishi) [inline src/features/ShopHomePage.js:31]: remove console.log and commented dummycode below

File: src/components/ProductsList.js [hunk 3]
```diff
@@ -1,14 +1,21 @@
 import React from "react";
 import { Link } from "react-router-dom";
 
-export default function ProductsList({ products }) {
+export default function ProductsList({ products, listType }) {
   return (
     <>
-      <h2>Products</h2>
+      {listType === "bestRated" ? (
```
Comment (woeishi) [inline src/components/ProductsList.js:7]: component is better reusable if the heading is not included. e.g. h2 might be the wrong level of heading.
this way the title can be independent of the lists loading state

**Review by woeishi — CHANGES_REQUESTED** [review]

File: src/features/ShopHomePage.js (hunk 2 above)
Comment (alexWaligorski) [reply src/features/ShopHomePage.js:31]: done! tysm.

**Pull request merged**